### DISCUSSION TITLE
Set scrollTo in a hook?

### DISCUSSION THREAD

**Discussion by misog** [PaperStrike/Pjax#337] [Q&A]
Hi, I would like to set scrollTo in a hook after HTTP request was created. Is it possible? Thank you.

**Comment by PaperStrike** [comment]
Could you share your use case for this? Currently if you would like to set the scroll position to a value that is variable on page switch, you can set `scrollTo` to `false` and add a custom `pjax:success` listener that controls the scroll position.

**Reply by misog** [reply]
I would like to scroll to top only if request was redirected. Other use case could be that scroll to top is enabled if [data-scroll-top] is present on the trigger element.

The problem with `pjax:success` approach is that it would scroll two times - once when handling the  `scrollTo` and second time by code in `pjax:success`. However that could create blinks. There should be way to set `scrollTo` on runtime and let the plugin to do the job of scrolling.

I think there is more general problem with this because although there are hooks they do not receive plugin config options and can not be used to achieve flexible behavior on runtime. For example, I have to create global variables to share data between hooks.

Workaround is to create two Pjax instances, ex. pjaxScroll and pjaxNoScroll and use:
```
    defaultTrigger: {
        exclude: 'a[data-scroll-top]',
    },
```
```
    defaultTrigger: {
        exclude: 'a:not([data-scroll-top])',
    },
```

**Reply by misog** [reply]
In [MoOx pjax](https://github.com/MoOx/pjax) I could do this and just handle response:
```
            pjax._handleResponse = pjax.handleResponse;

            pjax.handleResponse = function(responseText, request, href, options) {
                if (responseText && responseText.match("<html")) {
                    pjax._handleResponse(responseText, request, href, options);
                } else {
                    window.scrollTo(0, 0);
                    document.write(responseText || request.responseText);
                }
            }
```
